src/main/java/ch/njol/skript/lang/Trigger.java

```diff
@@ -48,27 +48,41 @@ public Trigger(final @Nullable File script, final String name, final SkriptEvent
 	}
 	
 	/**
-	 * Executes this trigger for certain event.
-	 * @param e Event.
-	 * @return false if an exception occurred
+	 * Executes this trigger for a certain event.
+	 * @param e The event to execute this Trigger with.
+	 * @return false if an exception occurred.
 	 */
-	public boolean execute(final Event e) {
+	public boolean execute(Event e) {
+		return execute(e, null);
+	}
+
+	/**
+	 * Executes this trigger for a certain event.
+	 * @param e The event ot execute this Trigger with.
+	 * @param afterExecution A Runnable to execute after {@link TriggerItem#walk(TriggerItem, Event)} has finished.
+	 * @return false if an exception occurred.
+	 */
+	public boolean execute(Event e, @Nullable Runnable afterExecution) {
```

Comment:
I don't think this one needs a Runnable, because the only thing being modified after walking the trigger is the local variables. I'm not sure if the better solution is to add a boolean whether or not to remove locals (or just a separate method), or to not modify anything in Trigger, instead requiring impls to use TriggerItem.walk directly.